code/game/machinery/mind_machine.dm

```diff
@@ -0,0 +1,846 @@
+		errorMessage = "Missing occupant"
+		occupantOne = null
+		occupantTwo = null
+		return
+	currentlySwapping = TRUE
+	lockedPods = TRUE
+	if((connectOne.icon_state != "mind_pod_closed") && (connectTwo.icon_state != "mind_pod_closed"))
+		playsound(connectOne, 'sound/machines/poddoor.ogg', 55, 1)
+		playsound(connectTwo, 'sound/machines/poddoor.ogg', 55, 1)
+		flick("mind_pod_closing",connectOne)
+		connectOne.icon_state = "mind_pod_closed"
+		flick("mind_pod_closing",connectTwo)
+		connectTwo.icon_state = "mind_pod_closed"
+	sleep(2 SECONDS)
+	playsound(connectOne, 'sound/effects/sparks4.ogg', 80, 1)
+	playsound(connectTwo, 'sound/effects/sparks4.ogg', 80, 1)
+	occupantScan = TRUE
+	S = occupantOne
+	occupantNameOne = S.name
+	switch(S.stat)
+		if(0)
+			occupantStatOne = "Alive"
+		if(1)
+			occupantStatOne = "Unconscious"
+		if(2)
+			occupantStatOne = "Dead"
+	if(!S.mind)
+		mindTypeOne = "Lower" //Simple mob
+	if(S.mind)
+		if(isrobot(S))
+			mindTypeOne = "Artificial" //Silicon player, obviously
+		else
+			mindTypeOne = "Higher" //Player controlled
+	if(isvampire(S) || isanycultist(S) || ischangeling(S) || ismalf(S))
+		mindTypeOne = "Shielded" //Mostly to fix spell bugs but also tinfoil
+	if((ishigherbeing(S)) || (ismonkey(S)))
+		var/mob/living/carbon/T = S
+		if(T.is_wearing_item(/obj/item/clothing/head/tinfoil))
+			mindTypeTwo = "Shielded"
+	S = occupantTwo
+	occupantNameTwo = S.name //and for pod two
+	switch(S.stat)
+		if(0)
+			occupantStatTwo = "Alive"
+		if(1)
+			occupantStatTwo = "Unconscious"
+		if(2)
+			occupantStatTwo = "Dead"
+	if(!S.mind)
+		mindTypeTwo = "Lower"
+	if(S.mind)
+		if(isrobot(S))
+			mindTypeTwo = "Artificial"
+		else
+			mindTypeTwo = "Higher"
+	if(isvampire(S) || isanycultist(S) || ischangeling(S) || ismalf(S))
+		mindTypeTwo = "Shielded"
+	if((ishigherbeing(S)) || (ismonkey(S)))
+		var/mob/living/carbon/T = S
+		if(T.is_wearing_item(/obj/item/clothing/head/tinfoil))
+			mindTypeTwo = "Shielded"
+	currentlySwapping = FALSE
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/swapOccupants(var/mob/living/M)
+	if(occupantScan != TRUE || lockedPods != TRUE || badSwap == TRUE)
+		return
+	if((!locate(occupantOne,connectOne.contents)) || (!locate(occupantTwo, connectTwo.contents)))
+		errorMessage = "Occupants missing" //sanity and wizards exist
+		occupantOne = null
+		occupantTwo = null
+		unlockPods()
+		return
+	if(bluespaceConduit == 0)
+		errorMessage = "Conduit requires charge"
+		return
+	if(mindTypeOne == "Shielded" || mindTypeTwo == "Shielded")
+		errorMessage = "Critical error, aborting"
+		spark(src)
+		spark(connectOne)
+		spark(connectTwo)
+		unlockPods()
+		return
+	if(occupantStatOne == "Dead" || occupantStatTwo == "Dead")
+		if(!soulShardSafety) //Secrets
+			errorMessage = "Living mind required"
+			return
+		soulShardSafety = FALSE
+		errorMessage = "Warning: Unknown internal reaction"
+	currentlySwapping = TRUE
+	bluespaceConduit -= 1
```

Comment:
I saw a += on monkeymen, so I looked for this. are you completely sure you want to like uhhh, use up their bluespace resources that early?

Reply:
Indeed. Unless the intent is that the bluespace resources *can* get wasted if it gets interrupted (because there seem to be plenty causes of interruption possible) then you should probably deduct later. There's plenty of situations in which you aren't refunded right now and I'm not sure that's intentional.

Also the lower lifeform thing should probably be an earlier check, yeah?